cfme/tests/v2v/test_v2v_migrations_ui.py

```diff
@@ -566,3 +569,74 @@ def _cleanup():
     view = navigate_to(migration_plan, "InProgress")
     assert not view.progress_card.is_plan_started(migration_plan.name)
     assert "no conversion host" in view.progress_card.get_error_text(migration_plan.name)
+
+
+@pytest.mark.uncollectif(lambda provider: provider.one_of(RHEVMProvider))
+@pytest.mark.parametrize("attribute", ["flavor", "security_group"])
+@pytest.mark.parametrize(
+    "mapping_data_vm_obj_single_datastore", [["nfs", "nfs", rhel7_minimal]], indirect=True)
+def test_v2v_custom_attribute(
+        request, appliance, provider, attribute, mapping_data_vm_obj_single_datastore):
+    """
+    Test V2V with custom attributes of openstack provider projects
+    Polarion:
+        assignee: sshveta
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
+        casecomponent: V2V
+        initialEstimate: 1/4h
+    """
+    infrastructure_mapping_collection = appliance.collections.v2v_infra_mappings
+    mapping_data = mapping_data_vm_obj_single_datastore.infra_mapping_data
+
+    # Changing target cluster to other than default(admin to qe-auto)
+    component = mapping_data["clusters"][0]
+    # Removed 'admin' (default project) from component
+    component.targets.pop()
+    # Add 'qe-auto' (different project) to component
+    component.targets.append(partial_match(provider.data.clusters[1]))
+    map_cluster = component.targets[0]
+    # create mapping with 'qe-auto' project
+    mapping = infrastructure_mapping_collection.create(**mapping_data)
+    src_vm_obj = mapping_data_vm_obj_single_datastore.vm_list[0]
+
+    # Selecting second flavor and security group selection since first is default
+    map_flavor = provider.data.flavors[1] if attribute == "flavor" else provider.data.flavors[0]
+    map_security_group = (provider.data.security_groups.admin[1]
+        if attribute == "security_group" else provider.data.security_groups.admin[0])
+
+    migration_plan_collection = appliance.collections.v2v_migration_plans
+    migration_plan = migration_plan_collection.create(
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()),
+        description="desc_{}".format(fauxfactory.gen_alphanumeric()),
+        infra_map=mapping.name,
+        osp_security_group=map_security_group,
+        osp_flavor=map_flavor,
+        target_provider=provider,
+        vm_list=mapping_data_vm_obj_single_datastore.vm_list,
+    )
+    assert migration_plan.wait_for_state("Started")
+    assert migration_plan.wait_for_state("In_Progress")
+    assert migration_plan.wait_for_state("Completed")
+    assert migration_plan.wait_for_state("Successful")
```

Comment:
Do we need to check each step of the way? Can we just assert success?

Reply:
We cannot @john-dupuy . Each of these are different methods .Only In_progress wait for plan to finish , rest all just checks plan status. So we have to go sequentially . 
Initially it was a single method that did everything but lot of tests requires only few of these states so made them individual methods.